crates/gossip/src/gossip.rs

```diff
@@ -279,32 +854,98 @@ mod test {
-            .send_module(
-                addr_1,
-                space.clone(),
-                MOD_NAME.to_string(),
-                serialize_gossip_message(K2GossipMessage {
-                    gossip_message: Some(GossipMessage::Initiate(
-                        K2GossipInitiateMessage {
-                            participating_agents: vec![],
-                            new_since: 0,
-                        },
-                    )),
-                })
-                .unwrap(),
+        let harness_2 = factory.new_instance().await;
+        harness_2.join_local_agent().await;
+        harness_2
+            .peer_store
+            .insert(vec![agent_info_1])
+            .await
+            .unwrap();
+
+        // Join extra agents for each peer. These will take a few seconds to be
+        // found by bootstrap. Try to sync them with gossip.
+        let secret_agent_1 = harness_1.join_local_agent().await;
+        let secret_agent_2 = harness_2.join_local_agent().await;
+
+        assert!(harness_1
+            .peer_store
+            .get(secret_agent_2.agent().clone())
+            .await
+            .unwrap()
+            .is_none());
+        assert!(harness_2
+            .peer_store
+            .get(secret_agent_1.agent().clone())
+            .await
+            .unwrap()
+            .is_none());
+
+        harness_2
+            .gossip
+            .initiate_gossip(agent_1.agent().clone())
+            .await
+            .unwrap();
+
+        harness_1
+            .wait_for_agent_in_peer_store(secret_agent_2.agent().clone())
+            .await;
+        harness_2
+            .wait_for_agent_in_peer_store(secret_agent_1.agent().clone())
+            .await;
+    }
+
+    #[tokio::test]
+    async fn two_way_op_sync() {
+        enable_tracing();
+
+        let space = test_space();
+        let factory = TestGossipFactory::create(space.clone()).await;
+        let harness_1 = factory.new_instance().await;
+        let agent_1 = harness_1.join_local_agent().await;
+        harness_1
+            .op_store
+            .process_incoming_ops(vec![MemoryOp::new(
+                Timestamp::now(),
+                vec![1; 128],
             )
+            .into()])
             .await
             .unwrap();
 
-        // TODO assert the outcome via the op store and peer metadata
+        let harness_2 = factory.new_instance().await;
+        harness_2.join_local_agent().await;
+        harness_2
+            .op_store
+            .process_incoming_ops(vec![MemoryOp::new(
+                Timestamp::now(),
+                vec![2; 128],
+            )
+            .into()])
+            .await
+            .unwrap();
+
+        harness_2
+            .gossip
+            .initiate_gossip(agent_1.agent().clone())
+            .await
+            .unwrap();
+
+        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+
+        // TODO fetch ops and assert that the op stores get updated.
```

Comment:
This is a special test where you run it and look for "Discovered op ids:" in the output :)

This test is working for me and op ids are being exchanged but without the fetch queue there's no way to retrieve the op data yet.